src/k8s-extension/azext_k8s_extension/partner_extensions/DataProtectionKubernetes.py

```diff
@@ -128,12 +143,27 @@ def Update(
         if configuration_settings is None:
             configuration_settings = {}
 
+        bsl_specified = False
         if len(configuration_settings) > 0:
             bsl_specified = self.__is_bsl_specified(configuration_settings)
             self.__validate_and_map_config(configuration_settings, validate_bsl=bsl_specified)
             if bsl_specified:
                 self.__validate_backup_storage_account(cmd.cli_ctx, resource_group_name, cluster_name, configuration_settings)
 
+        # this step is for brownfield migrating to AAD
+        if configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_USE_AAD) is not None and configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_USE_AAD).lower() == "true":
+            logger.warning("useAAD flag is set to true. Please provide extension MSI Storage Blob Data Contributor role to the storage account.")
+
+        # SA Uri not provided in user input, and not populated in the original extension, we populate it.
```

Comment:
Consider the following case: 
Earlier BSL uri of SA1 was present. Now, an update comes, where only new BSL details of SA2 (bucket, sa, rg, sub) are provided. From what I can see, after the update, the BSL details will belong to SA2, but Uri will point to SA1, which will lead to failures.